elements/includes/creative_commons.inc

```diff
@@ -266,57 +701,132 @@ function xml_form_elements_creative_commons_ajax(&$form, &$form_state) {
  * @param string $commercial
  *   'y' to allow 'n' to disallow.
  * @param string $derivatives
- *   'y' to allow 'n' to disallow.
+ *   'y' to allow 'n' to disallow, 'sa' to share alike.
  * @param string $jurisdiction
  *   Legal jurisdiction code.
+ * @param bool $disabled
+ *   TRUE returns FALSE, FALSE (default) performs request.
+ * @param string $version
+ *   Specify (earlier) version (blank (default) for newest).
+ * @param string $license
+ *   Get licenses other than 'by' (default): 'zero', 'mark', 'sampling+'.
  *
  * @return mixed
  *   SimpleXMLElement the return from the REST API.
  *   FALSE if failed request.
  */
-function xml_form_elements_get_creative_commons($commercial, $derivatives, $jurisdiction, $disabled = FALSE) {
+function xml_form_elements_get_creative_commons($commercial, $derivatives, $jurisdiction, $disabled = FALSE, $version = '', $license = 'by') {
+  // Do not make request if disabled. Otherwise, either get an up-to-date
+  // license if $version is empty, or get license details if $version is
+  // specified.
   if ($disabled) {
     return FALSE;
   }
-  else {
-    $response = drupal_http_request(url(
-      'http://api.creativecommons.org/rest/1.5/license/standard/get',
-      array(
-        'query' => array(
-          'commercial' => $commercial,
-          'derivatives' => $derivatives,
-          'jurisdiction' => $jurisdiction,
-        ),
-      )
-    ));
-    if ($response->code != 200) {
-      error_log("xml_form responded {$response->code}");
-      return FALSE;
-    }
-    return simplexml_load_string($response->data, 'SimpleXMLElement');
+  elseif ($version != '') {
+    return xml_form_elements_get_creative_commons_from_uri(xml_form_elements_creative_commons_value($commercial, $derivatives, $jurisdiction, $disabled, $version, $license));
```

Comment:
@DiegoPino Mmmhhh... I am not sure if I know exactly how to solve this, but I will give it some thought.

The function names are, indeed, based on the legacy names (also the order of parameters) in order not to break people's code. Regarding the _misnomer_, would it be o.k. to call the function `xml_form_elements_creative_commons_get_api_response_from_cache()`, even though it might perform an API request, or does the problem you are seeing go deeper? (Also, I do not quite understand what _else_ they are responding, besides `FALSE` on failure - or did I miss your point?)